### PR TITLE
docs: split documentation into individual pages

### PR DESCRIPTION
**Pull request by hay-kot**
## What type of PR is this?

- documentation

## What this PR does / why we need it:

Improves documentation by breaking up pages into individual sections/pages making it easier to navigate _(In my opinion)_

## Special notes for your reviewer:

This PR extends some of the work done in #1343 by breaking up the manuals into individual pages to improve navigate and ability to find things. I've only done the v1 docs, but will finish the v2 if I can get some confirmation that this is something you folks are interested in. 

<details>
<summary>
Before
</summary>

CleanShot 2022-08-14 at 14 59 54

</details>

<details>
<summary>
After
</summary>

CleanShot 2022-08-14 at 15 02 21

</details>

Preview at -> https://hay-kot.github.io/cli/v1/examples/generated-help-text/

### CODE REVIEW

**Comment by meatballhat** [comment]
@hay-kot Amazing work! Thank you so much! I would love to see this for the v2 guide, too ❤️

**Comment by hay-kot** [comment]
Thanks!

Just pushed the changes version v2 docs. 

I also made two changes to improve the search results.

1. Added a `v1` and `v2` tag to applicable pages which display in search. Before results from v1 vs v2 were indistinguishable
2. I "boosted" all v2 pages to hopefully put those at the top of the results when the pages are nearly duplicate.  

CleanShot 2022-08-14 at 16 08 28

**Review by dearchap — APPROVED** [review]
Amazing. Thank you so much

**Comment by hay-kot** [comment]
Ahhh, looks like an unintended side effect of splitting the documentation was that the gfmrun test now fails. 😞

I implemented a `--walk` flag that will walk the directory and collect all the markdowns and pass them along to the tool. There is a discrepancy between the count and the results from the tool. I didn't see anything obvious, but could use another set of eyes, it's probably something silly.

Here's that snippet. 

https://github.com/urfave/cli/pull/1453/files#diff-f2ae1f6d37751ba11f66da69f40d1c3076bb1f2246878641080d5855034881b3R259-R282

**Review by dearchap — DISMISSED** [review]

File: internal/build/build.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func GfmrunActionFunc(cCtx *cli.Context) error {
 
 	if walk {
 		// Walk the directory and find all markdown files.
-		err := filepath.Walk(top, func(path string, info os.FileInfo, err error) error {
+		err := filepath.Walk(filename, func(path string, info os.FileInfo, err error) error {
```
Comment (dearchap) [inline internal/build/build.go:235]: cant we rename filename to like dirpath or something to make it clear that its a dir ? or even as root

**Review by dearchap — APPROVED** [review]
Thank you @hay-kot

**Pull request merged**